src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -1931,6 +1931,15 @@ static void window_ride_construction_mouseup_demolish(rct_window* w)
         const rct_preview_track* trackBlock = ted.Block;
         newCoords->z = (tileElement->GetBaseZ()) - trackBlock->z;
         gGotoStartPlacementMode = true;
+
+        // When flat rides are deleted window should be reset
+        const auto rideId = w->rideId;
+        auto ride = get_ride(rideId);
+        if (ride->GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_FLAT_RIDE)
+            && !ride->GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_IS_SHOP))
```

Comment:
I think you can drop the `!ride->GetRideTypeDescriptor().HasFlag(RIDE_TYPE_FLAG_IS_SHOP)` check here - even if shops would somehow end up in place entrance/exit mode, you _want_ them to snap out of it.

Reply:
Very good. I have already tested your changes, they seem to solve the problem.